src/widget/wtracktableview.cpp

```diff
@@ -1373,6 +1401,31 @@ QList<TrackId> WTrackTableView::getSelectedTrackIds() const {
     return trackIds;
 }
 
+void WTrackTableView::setSelectedTracks(QList<TrackId> trackIds) {
```

Comment:
This parameter should be passed as const-ref to avoid a deep copy of the list. Unfortunately the range-based for loops do not work well with Qt containers. Declaring the loop variable as const-ref is not enough. The container itself must be const! There is already a link in our [coding guidelines](https://www.mixxx.org/wiki/doku.php/coding_guidelines#range-based_for_loops).

We have this issue at many places and I have done this wrong until recently. Checking the code with [clazy](https://github.com/KDE/clazy) could reveal those hidden performance killers.

Reply:
I will fix this after merge

Reply:
I already did in #1427

Reply:
included in 187ec3d5fa231f2ee1259fe724ca50386a8ff9fc by @Be-ing